e2e/kube.go

```diff
@@ -0,0 +1,72 @@
+package e2e_test
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type kubeclient struct {
+	dynamic client.Client
+	typed   kubernetes.Interface
+	rest    *rest.Config
+}
+
+func newKubeclient(config *rest.Config) (*kubeclient, error) {
+	dynamic, err := client.New(config, client.Options{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create dynamic kubeclient: %q", err)
+	}
+
+	restClient, err := rest.RESTClientFor(config)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create rest kube client: %q", err)
+	}
+
+	typed := kubernetes.New(restClient)
+
+	return &kubeclient{
+		dynamic: dynamic,
+		typed:   typed,
+		rest:    config,
+	}, nil
+}
+
+func getClusterKubeClient(ctx context.Context, cloud *azureClient, config *suiteConfig) (*kubeclient, error) {
+	data, err := getClusterKubeconfigBytes(ctx, cloud, config.resourceGroupName, config.clusterName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get cluster kubeconfig bytes")
+	}
+
+	restConfig, err := clientcmd.RESTConfigFromKubeConfig(data)
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert kubeconfig bytes to rest config")
+	}
+	restConfig.NegotiatedSerializer = serializer.WithoutConversionCodecFactory{CodecFactory: scheme.Codecs}
+	restConfig.APIPath = "/api"
```

Comment:
I swear this is wrong but it works _shrugs_

should be /api for v1.Node,Pod, but /apis for DS. dynamic client works as expected for DS, didn't test typed client yet.